### DISCUSSION TITLE
Clear hash map

### DISCUSSION THREAD

**Discussion by travisdoor** [nothings/stb#1227] [Q&A] [answered]
Hi, is there any way how to clear the hash map content and reuse already allocated memory?

**Comment by nothings** [comment, marked as answer]
nope

**Reply by nothings** [reply]
to clarify:

(1) while dynamic arrays are relatively lightweight and such an operation would make sense for them, hash tables are not very lightweight, and stb_ds.h hash tables are not maximally efficient in the first place; they make multiple compromises to provide a convenient API, and are intended for use where you really just want the O(1) hash table performance, but don't care about max performance.

(2) stb_ds hash tables automatically double and halve their size as you add and delete elements. for this reason, memory reuse doesn't make very much sense, as the hash table part would naturally shrink and grow automatically anyway; it would require extra code to try to reuse the table without automatically shrinking.
